sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ReceiverManager.cs

```diff
@@ -144,6 +144,13 @@ protected async Task ProcessOneMessageWithinScopeAsync(ServiceBusReceivedMessage
 
             try
             {
+                if (message.LockedUntil <= DateTimeOffset.UtcNow)
```

Comment:
The locked until time being the past doesn't necessarily mean that the lock is lost. If the lock is truly lost, then abandoning the message will fail. 

I think we should fix the root cause by queueing up the prefetched messages. A recent change was to the ReceiveMessagesAsync method was made so that a TimeSpan of zero can be passed when prefetch is enabled. By passing zero, we can drain the local queue and start renewing the message locks ahead of time.

Reply:
In testing (prior to this PR) I too was expecting that calling Abandon would fail. 
However in testing without calling Abandon, no messages (new or existing) would be received by the ServiceBusReceiver. When calling Abandon on a message internally the AmqpReceiver holds an expiring dictionary of messages, and if the messages lock token has expired it will skip the remote Abandon Request/Response, and instead just dispose the message from the prefetched list.

I tried to find where to just clear all the prefetched messages in one go, but there does not appear to be that functionality (atleast within this library), is the TimeSpan.Zero change your referring to committed to this repo?

Reply:
> In testing (prior to this PR) I too was expecting that calling Abandon would fail.
However in testing without calling Abandon, no messages (new or existing) would be received by the ServiceBusReceiver. When calling Abandon on a message internally the AmqpReceiver holds an expiring dictionary of messages, and if the messages lock token has expired it will skip the remote Abandon Request/Response, and instead just dispose the message from the prefetched list

The expiring set is specific to deferred messages that are received so I don't think it applies here. More likely, the service is still treating the message as locked even though the locked time has already passed - I believe it provides a bit of a buffer. 

> I tried to find where to just clear all the prefetched messages in one go, but there does not appear to be that functionality (atleast within this library), is the TimeSpan.Zero change your referring to committed to this repo?

It was a change in this repo, and in the AMQP repo that allowed for TimeSpan.Zero to not go to the service:
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/src/Receiver/ServiceBusReceiver.cs#L263

https://github.com/Azure/azure-amqp/blob/cf2e8c4d08ebbcd9bd3291b666381c68e71c6ba7/Microsoft.Azure.Amqp/Amqp/ReceivingAmqpLink.cs#L113

https://github.com/Azure/azure-amqp/blob/cf2e8c4d08ebbcd9bd3291b666381c68e71c6ba7/Microsoft.Azure.Amqp/Amqp/ReceivingAmqpLink.cs#L203

The net result of these changes should allow us to call Receive and pass in the PrefetchCount as the maxMessages along with TimeSpan.Zero as the maxWaitTime.